File: egressos/src/main/java/com/labprog/egressos/service/FaixaSalarioService.java
package com.labprog.egressos.service;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.labprog.egressos.model.FaixaSalario;
import com.labprog.egressos.model.repository.FaixaSalarioRepo;

@Service
public class FaixaSalarioService {
    @Autowired
    FaixaSalarioRepo repository;

    public List<FaixaSalario> buscar() {
        return repository.findAll();
    }

    public int quantidadeEgressoPorFaixaSalario(FaixaSalario fs){
        int quantidade = repository.quantidadeEgressoPorFaixaSalario(fs);
        return quantidade;
    }

    public FaixaSalario obterFaixaSalarioPorDescricao(String descricao){
        return repository.obterFaixaSalarioPorDescricao(descricao);
    }
}

File: egressos-view/src/services/FaixaSalarioService.js
import ApiService from './ApiService'
class FaixaSalarioService extends ApiService {
    constructor () {
        super('api/faixa_salario')
    }
    obterFaixasSalariais() {
        return this.get('/buscar')
    }
    quantidaDeEgressosPorFaixaSalarial(id) {
        return this.get(`/quantidade_egresso_por_faixa_salario/${id}`)
    }
}
export default FaixaSalarioService